prospero-common/src/main/java/org/wildfly/prospero/actions/ProvisioningAction.java

```diff
@@ -57,26 +60,23 @@ public ProvisioningAction(Path installDir, MavenSessionManager mavenSessionManag
     }
 
     /**
-     * Installs feature pack defined by {@code fpl} in {@code installDir}. If {@code fpl} doesn't include version,
-     * the newest available version will be used.
+     * Provision installation according to given ProvisioningDefinition.
      *
-     * @param config
-     * @param channels
-     * @throws ProvisioningException
-     * @throws MetadataException
+     * @param provisioningDefinition prospero provisioning definition
      */
-    public void provision(ProvisioningConfig config, List<Channel> channels) throws ProvisioningException, OperationException {
-        doProvision((pm,options)->pm.provision(config, options), enforceChannelNames(channels));
-    }
+    public void provision(ProvisioningDefinition provisioningDefinition)
```

Comment:
I think I'd rather keep this using ProvisioningConfig and channels directly rather then ProvisioningDefinition

Reply:
OK, I wanted to do channel resolution inside the `ProvisioningAction` class, but it turns out it's OK to do it in the `InstallCommand` as well.